File: C/7.cpp
#include <iostream>
#include <string>
using namespace std;

long min_del(long num, int k)
{
	if(k == 0)
		return num;
	string snum = to_string(num);
	string d_num = snum;
	int l = snum.size();
	long min = stol(d_num.erase(0, 1));
	for(int i = 1; i<l;i++)
	{
		d_num = snum;
		if(stol(d_num.erase(i, 1)) < min)
			min = stol(d_num);
	}
	long res = min_del(min, k-1);	
	return res;
}

int main()
{
	int n, k;
	long num;
	cin >> n >> k;
	cin >> num;
	cout << min_del(num, k);
}

File: python/7.py
#!/usr/bin/env python
# coding: utf-8

# In[57]:


def erase(s, idx):
    return s[:idx] + s[idx + 1:]

def min_del(num, k):
	if(k == 0):
		return num
	snum = str(num)
	l = len(snum)
	min = int(erase(snum, 0))
	for i in range(1, l):
		if int(erase(snum, i)) < min:
			min = int(erase(snum,i))
	res = min_del(min, k-1)	
	return res;


# In[ ]:


nk = list(map(int, input().split()))
n = nk[0]
k = nk[1]
num = int(input())
print(min_del(num, k))

